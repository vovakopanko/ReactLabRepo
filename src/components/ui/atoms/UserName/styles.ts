import { colors } from "../../../../styles/palette";

import styled from "styled-components";

export const StyleItem = styled.ul`
  font-size: 18px;
  color: ${colors.WHITE};
  list-style-type: none;
  cursor: pointer;
  &:hover {
    color: ${colors.PURPURE};
  }
  padding: 0 20px;
`;

export const TitleItem = styled.span`
  padding-right: 10px;
`;
