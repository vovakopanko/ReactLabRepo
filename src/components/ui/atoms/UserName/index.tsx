import { colors } from "../../../../styles/palette";
import styled from "styled-components";

type Props = {
  userName: string;
};

const UserName = ({ userName }: Props) => {
  return (
    <StyleItem>
      <TitleItem>{userName}</TitleItem>
    </StyleItem>
  );
};

export default UserName;

const StyleItem = styled.ul`
  font-size: 18px;
  color: ${colors.WHITE};
  list-style-type: none;
  cursor: pointer;
  &:hover {
    color: ${colors.PURPURE};
  }
  padding: 0 20px;
`;

const TitleItem = styled.span`
  padding-right: 10px;
`;
