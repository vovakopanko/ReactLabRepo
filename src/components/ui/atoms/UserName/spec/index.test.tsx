import renderer from "react-test-renderer";
import UserName from "../index";

describe("render snapShot", () => {
  it("should return UserName layout component", () => {
    const tree = renderer.create(<UserName userName={"Test User"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
