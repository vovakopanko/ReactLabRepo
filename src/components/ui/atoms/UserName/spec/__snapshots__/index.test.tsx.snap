// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapShot should return UserName layout component 1`] = `
<ul
  className="sc-bdvvtL cAezNv"
>
  <span
    className="sc-gsDKAQ igZSXo"
  >
    Test User
  </span>
</ul>
`;
